src/backoff.ts

```diff
@@ -19,28 +19,50 @@ export async function backOff<T>(
 
 class BackOff<T> {
   private attemptNumber = 0;
+  private isAborted = false;
 
   constructor(
     private request: () => Promise<T>,
     private options: IBackOffOptions
   ) {}
 
+  private _cancelWhenAborted() {
+    const { signal } = this.options;
+
+    if (!signal) return;
+
+    if (signal.aborted) {
+      this.isAborted = true;
+      return;
+    }
+
+    signal.addEventListener(
+      "abort",
+      () => {
+        this.isAborted = true;
+      },
+      { once: true }
+    );
+  }
+
   public async execute(): Promise<T> {
-    while (!this.attemptLimitReached) {
+    this._cancelWhenAborted();
+
+    while (!this.isAborted && !this.attemptLimitReached) {
       try {
         await this.applyDelay();
         return await this.request();
       } catch (e) {
         this.attemptNumber++;
         const shouldRetry = await this.options.retry(e, this.attemptNumber);
 
-        if (!shouldRetry || this.attemptLimitReached) {
+        if (!shouldRetry || this.isAborted || this.attemptLimitReached) {
```

Comment:
I think it's not correct to do an abort in between executions -  same behavior could be achived in user-defined retry method. Instead setTimeout located  [here](https://github.com/coveooss/exponential-backoff/blob/master/src/delay/delay.base.ts#L10), should be cleared.